src/render/src/encoder.rs

```diff
@@ -84,6 +90,41 @@ pub struct Submit<B: Backend, C> {
     pub(crate) pool: PoolDependency<B, C>
 }
 
+// TODO: coalescing?
+struct ImageStates {
+    states: Vec<core::image::State>,
```

Comment:
We could probably just add the level/slice into the hashmap key instead of introducing this struct.
Do you think the struct gives us an advantage?

Reply:
I was wondering if it would be worth it to coalesce the state slices instead of always having a state for each level/layer. Otherwise, I don't see any advantage.

Reply:
Ok, let's have it this way for now, will reconsider later.